modules/passkey/test/4337/WebAuthn.spec.ts

```diff
@@ -9,7 +9,7 @@ import {
 } from '@safe-global/safe-4337/src/utils/userOp'
 import { chainId } from '@safe-global/safe-4337/test/utils/encoding'
 import { Safe4337 } from '@safe-global/safe-4337/src/utils/safe'
-import { WebAuthnCredentials, decodePublicKey, encodeWebAuthnSignature } from '../utils/webauthn'
+import { WebAuthnCredentials, decodePublicKey, encodeWebAuthnSignature } from '../../src/utils/webauthn'
```

Comment:
Hmm, `webauthn.ts` has two parts:
1. a shim that should **only** be used for testing
2. utilities for interacting with WebAuthn API (extracting signatures, public key, etc.)

These need to be split and we cannot just move the whole module.

Reply:
So, it has to be mainly divided as?

`test/utils`:
- Dummy values
- `WebAuthnCredentials`

`src/utils`:
- `decodeSignature, getSignatureBytes, encodeWebAuthnSignature`
- `decodeClientDataFields, decodePublicKey, base64UrlEncode, encodeWebAuthnSigningMessage`

Reply:
Dummy values should be in `src`, they are used for signature gas estimation.

Reply:
Ok, so just `WebAuthnCredentials` should be in `test/utils`, the rest in `src/utils`, right?

Reply:
Yup, just the credential shim (and related types). I think `src` should even use the builtin TypeScript types for the WebAuthn API.

Reply:
The first part is done, the second part might need more clarification. Did you mean about using `bigint` instead of `BigNumberish`, or something else?

Reply:
No, TypeScript has built-in types for the `navigator.credential...` API. I think you already did this since there are no type errors and you didn't copy over the interface definitions.